amazon-kinesis/1.0/modules/ROOT/pages/amazon-kinesis-connector-reference.adoc

```diff
@@ -0,0 +1,522 @@
+|===
+| Field | Type | Description | Default Value | Required
+| Fails Deployment a| Boolean | When the application is deployed, a connectivity test is performed on all connectors. If set to `true`, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy. |  |
+| Reconnection Strategy a| * <<Reconnect>>
+* <<ReconnectForever>> | The reconnection strategy to use. |  |
+|===
+
+[[Reconnect]]
+=== Reconnect
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Frequency a| Number | How often in milliseconds to reconnect |  |
+| Count a| Number | How many reconnection attempts to make. |  |
+|===
+
+[[ReconnectForever]]
+=== Reconnect Forever
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Frequency a| Number | How often in milliseconds to reconnect |  |
+|===
+
+[[RecordAttributes]]
+=== Record Attributes
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Aggregated a| Boolean |  |  |
+| Approximate Arrival Timestamp a| String |  |  |
+| Encryption Type a| String |  |  |
+| Partition Key a| String |  |  |
+| Partition Key a| String |  |  |
+| Sequence Number a| String |  |  |
+| Shard Id a| String |  |  |
+| Sub Sequence Number a| Number |  |  |
+|===
+
+[[RedeliveryPolicy]]
+=== Redelivery Policy
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Max Redelivery Count a| Number | The maximum number of times a message can be redelivered and processed unsuccessfully before triggering a process-failed message |  |
+| Message Digest Algorithm a| String | The secure hashing algorithm to use. |  SHA-256|
+| Message Identifier a| <<RedeliveryPolicyMessageIdentifier>> | The strategy used to identify the messages. |  |
+| Object Store a| ObjectStore | The object store that stores the redelivery counter for each message. |  |
+|===
+
+[[RedeliveryPolicyMessageIdentifier]]
+=== Redelivery Policy Message Identifier
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Use Secure Hash a| Boolean | Whether to use a secure hash algorithm to identify a redelivered message. |  |
+| Id Expression a| String | One or more expressions to use to determine when a message was redelivered. This property can be set only if the value of the *Use Secure Hash* field is `false`. |  |
+|===
+
+[[ShardConfig]]
+=== Shard Config
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| Shard Id a| String | The shard identifier. |  | x
+| Shard Initial Position a| <<ShardInitialPosition>> | The shard initial position. |  | x
+|===
+
+[[ShardInitialPosition]]
+=== Shard Initial Position
+
+[%header,cols="20s,25a,30a,15a,10a"]
+|===
+| Field | Type | Description | Default Value | Required
+| At Sequence Number a| String |  |  |
```

Comment:
at SN: `This option ensures that listeners will listen from a specific sequence number.`
After SN: `The option ensures that listeners will listen after a specific sequence number.`
timestamp: `This option ensures that listeners will listen from a specific date and time.`
absolute position: `This option ensures that the listener will listen from the latest or oldest records.`